### PR TITLE
Sort, filters, readme

### LINKED ISSUE #74: [FEATURE] Toggle to only show favorite courses
This is a feature request from Marleen Denert who has a lot of courses in the sidebar and would like a way to show only the relevant ones in a logical order. Some courses have the same name but are from a different academic year.

* Setting (toggle) to only show favorite courses (off by default). This would resemble the sidebar on dodona.ugent.be (but alphabetically):
image
* Setting (toggle) to sort by academic year (descending order) in the first place and sort alphabetically second.

### LINKED ISSUE #75: [FEATURE] Add new filters to readme
The filters added in #74 can be a bit confusing to new users, add an example in `README.md` that explains this a bit clearer.

### PR DESCRIPTION
**Pull request by stijndcl**
Add option to sort courses: Fixes #74
Add option to filter courses by year & by course id: Fixes #74 
Update `README.md` to explain how this system works: Fixes #75

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -50,6 +50,26 @@ Once you've solved the exercise, it's easy to submit your solution. Execute the
...
+
+You can apply filters to _only_ show courses of certain academic years, or courses with certain id's, by using the `dodona.treeview.courses` and `dodona.treeview.years` fields. Both of these filters can be applied _at the same time_, and they will count as an **AND** filter, meaning that only courses meeting both filters will be shown.
+
+Academic years should be supplied using the `YYYY` format, and be separated by comma's. For example, to only show courses from `2016-2017` and `2020-2021`, you would use `2016,2020`.
+
+![FilterYears](assets/filter-years-example.png)
+
+Course id's can be found at the end of a course's URL, and should be separated by comma's. For example, to only show courses with id's `123` and `456`, you would use `123,456`.
+
+![FilterCOurses](assets/filter-courses-example.png)
```
Comment (stijndcl) [inline README.md:69]:
```suggestion
![FilterCourses](assets/filter-courses-example.png)
```

File: README.md [hunk 2]
```diff
@@ -50,6 +50,26 @@ Once you've solved the exercise, it's easy to submit your solution. Execute the
...
+
+When subscribed to a lot of courses, it can often become difficult to find the relevant ones. In case you are a teacher, it could also happen that you want to give a presentation to your students, but don't want to flash your exams and tasks in the sidebar.
+
+You can apply filters to _only_ show courses of certain academic years, or courses with certain id's, by using the `dodona.treeview.courses` and `dodona.treeview.years` fields. Both of these filters can be applied _at the same time_, and they will count as an **AND** filter, meaning that only courses meeting both filters will be shown.
+
+Academic years should be supplied using the `YYYY` format, and be separated by comma's. For example, to only show courses from `2016-2017` and `2020-2021`, you would use `2016,2020`.
+
+![FilterYears](assets/filter-years-example.png)
+
+Course id's can be found at the end of a course's URL, and should be separated by comma's. For example, to only show courses with id's `123` and `456`, you would use `123,456`.
```
Comment (thepieterdc) [inline README.md:67]: ' is not a plural thing in English

```suggestion
Course ids can be found at the end of a course's URL, and should be separated by commas. For example, to only show courses with id's `123` and `456`, you would use `123,456`.
```

File: package.json [hunk 3]
```diff
@@ -100,6 +100,33 @@
...
+            "Academic year first (descending), alphabetic second",
+            "Alphabetic first (ascending), academic year second",
+            "Alphabetic first (descending), academic year second"
+          ],
+          "description": "Specifies how the courses in the Tree View should be sorted."
+        },
+        "dodona.treeview.years": {
+          "type": "string",
+          "default": "",
+          "markdownDescription": "Specifies academic years of which courses should be visible, leave empty if no courses should be hidden. Academic years should be supplied using the `YYYY` format, and be separated by comma's."
```
Comment (thepieterdc) [inline package.json:124]:
```suggestion
          "markdownDescription": "Specifies academic years of which courses should be visible, leave empty if no courses should be hidden. Academic years should be supplied using the `YYYY` format, and be separated by commas."
```

File: package.json [hunk 4]
```diff
@@ -100,6 +100,33 @@
...
+        },
+        "dodona.treeview.courses": {
+          "type": "string",
+          "default": "",
+          "markdownDescription": "Specifies which course id's should be visible, leave empty if no courses should be hidden. Course id's should be separated by comma's."
```
Comment (thepieterdc) [inline package.json:129]:
```suggestion
          "markdownDescription": "Specifies which course ids should be visible, leave empty if no courses should be hidden. Course id's should be separated by commas."
```

File: src/treeView/dataProvider.ts [hunk 5]
```diff
@@ -53,4 +55,77 @@ export default class RootDataProvider
         // TODO optimise this to not redraw the entire tree.
         this._onDidChangeTreeData.fire(undefined);
     }
+
+    sortCourses(courses: Course[]): Course[] {
```
Comment (thepieterdc) [inline src/treeView/dataProvider.ts:59]: Can be static

File: src/treeView/items/courseTreeItem.ts [hunk 6]
```diff
@@ -3,6 +3,7 @@ import { ProviderResult, TreeItemCollapsibleState } from "vscode";
 import execute from "../../api/client";
 import { AbstractTreeItem } from "./abstractTreeItem";
 import { SeriesTreeItem } from "./seriesTreeItem";
+import { Console } from "console";
```
Comment (thepieterdc) [inline src/treeView/items/courseTreeItem.ts:6]: Cleanup import

**Review by thepieterdc — APPROVED** [review]

**Pull request merged**